### ISSUE TITLE
Add support for pcap-ng format

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1912]
When merging 2 files it can read individually, airodump-ng fails to read it and displays `"out.pcap" isn't a pcap file (expected TCPDUMP_MAGIC).`. tcpdump is able to read the file.

How to reproduce:
```
mergecap -w out.pcap test/wps2.0.pcap test/wep_64_ptw.cap
airodump-ng -r out.pcap
```

**Comment by ChrisLundquist** [comment]
FWIW you can *sometimes* use `mergcap -F pcap ...` to force the output file to be in `pcap` format.
`mergecap` will fail though if your inputs use new features like multiple interfaces, which will happen if you try to octomerge N (more than 2) pcap files.

**Comment by joswr1ght** [comment]
I would also like to see pcapng support in Aircrack-ng. I end up converting a lot of pcapng files to pcap `tshark -r foo.pcapng -w foo.pcap -F libpcap` but native pcapng support would be cool.

**Comment by ZerBea** [comment]
Moving from cap to pcapng would be a big improvement.

Some "good" reasons to move to pcapng:
https://wiki.wireshark.org/Development/PcapNg
https://cloudshark.io/articles/5-reasons-to-move-to-pcapng/

struct:
https://pcapng.github.io/pcapng/

BTW:
converting pcapng to cap isn't a good solution, because all additional information get lost e.g.:
hcxdumptool pcapng option codes (Section Header Block)
ENTERPRISE NUMBER
0x2a, 0xce, 0x46, 0xa1

MAGIC NUMBER
0x2a, 0xce, 0x46, 0xa1, 0x79, 0xa0, 0x72, 0x33,
0x83, 0x37, 0x27, 0xab, 0x59, 0x33, 0xb3, 0x62,
0x45, 0x37, 0x11, 0x47, 0xa7, 0xcf, 0x32, 0x7f,
0x8d, 0x69, 0x80, 0xc0, 0x89, 0x5e, 0x5e, 0x98

OPTIONCODE_MACMYORIG
0xf29a (6 byte)

OPTIONCODE_MACMYAP
0xf29b (6 byte)

OPTIONCODE_RC
0xf29c (8 byte)

OPTIONCODE_ANONCE
0xf29d (32 byte)

OPTIONCODE_MACMYSTA
0xf29e (6 byte)

OPTIONCODE_SNONCE
0xf29f (32 byte)

OPTIONCODE_WEAKCANDIDATE
0xf2a0 (32 byte)

OPTIONCODE_GPS
0xf2a1 (max 128 byte)

**Comment by geraldcombs** [comment]
The pcapng specification also defines a [decryption secrets block](https://ietf-opsawg-wg.github.io/draft-ietf-opsawg-pcap/draft-ietf-opsawg-pcapng.html#name-decryption-secrets-block). Block types for 802.11 decryption aren't currently defined, but they shouldn't be too difficult to add.

**Comment by Mister-X-** [comment]
That would be awesome. I'm assuming we'll have similar options as in the decryption options in Wireshark for 802.11 (WEP, WPA passphrase/PMK, TK, MSK)?

**Comment by ZerBea** [comment]
Correct. You can store PSK or PMK here.
If the draft come into force there is no longer need to store this information in pcapng comment fields.
That is a huge improvement.

BTW:
Any time line when aircrack-ng move from cap to pcapng?

**Comment by Mister-X-** [comment]
No timeline yet. Prior to that, we need to allow the radiotap headers to be passed back by osdep (which it currently strips) so they can be saved in the PCAP.

**Comment by ZerBea** [comment]
That is a great idea and will be a nice feature. We can take a lot of information from the radiotap header.

BTW:
On TX side, I started some tests to reduce the radiotap header. Instead of setting BITRATE via radiotap header I'll do it via NL80211 every time I change the channel. It can safely be done by a single NL80211 message:
NL80211_ATTR_IFINDEX
NL80211_ATTR_WIPHY_FREQ
NL80211_ATTR_CHANNEL_WIDTH
NL80211_ATTR_WIPHY_CHANNEL_TYPE
NL80211_ATTR_CENTER_FREQ1
NL80211_ATTR_TX_RATES
and the interface will change frequency, set channel width, set channel type, set center frequency and tx rate.

**Comment by Mister-X-** [comment]
From a performance perspective, isn't that slower?

**Comment by ZerBea** [comment]
No, I don't think it is slower. I still have to set a hold time to stay long enough on a channel, even though I use NL80211.
I don't use libnl, libn-tiny or any other library to talk to the driver. hcxlabtool generate a NL80211 message and send it direct via NL80211 protocol to the kernel. That is fast enough even on a Raspberry Pi.

hcxlabtool/hcxdumptool are able to request a PMKID from an ACCESS POINT (AP).
On 2.4GHz we can use 1MB/sec during the entire AUTHENTICATION. On 5GHz and above we have to set bitrate to 6MB/sec. I tested this running hcxlabtool against my target AP. It doesn't respond on 5GHz if we try an AUTHENTICATION running 1MB/sec, only.
Requesting an EAPOL M2 from a CLIENT (I love unencrypted M2 messages), it looks like  there is no 6MB/sec limitation. If we announce via IE_TAG that we can do 1MB/sec, on 5GHz all my test CLIENTs authenticate running 1MB/sec.

BTW:
I moved everything from WIRELESS EXTENSIONS and ETHTOOL (hcxdumptool is an ioctl system call dinosaur and it is time to retire it) to RTNETLINK and NL80211 (hcxlabtool) :
set interface up and down via RTNETLINK
get virtual MAC and hardware MAC via RTNETLINK
set MONITOR MODE via NL80211
set channel, bandwidth and center frequency via NL80211

Next step is to get benefit of ACTIVE MONITOR MODE. Unfortunately the documentation of this mode is poor.

I noticed that you can retrieve every information about the interface (capabilities and status) by RTLINK and NL80211 and you can set whatever you want via RTNETLINK and NL80211. Only a few lines of code and you take full control over the interface.
